src/Combinatorics/Graphs/functions.jl

```diff
@@ -108,6 +170,63 @@ function add_edge!(g::Graph{T}, source::Int64, target::Int64) where {T <: Union{
   return n_edges(g) == old_nedges + 1
 end
 
+@doc raw"""
+    add_directed_edge!(g::MixedGraph, s::Int64, t::Int64)
+
+Add edge `(s,t)` to the dirceted component of the graph `g`.
+Return `true` if a new edge `(s,t)` was added, `false` otherwise.
+
+# Examples
+```jldoctest
+julia> g = mixed_graph(2)
+Mixed graph with 2 nodes and no edges
+
+julia> add_directed_edge!(g, 1, 2)
+true
+
+julia> add_directed_edge!(g, 1, 2)
+false
+
+julia> n_edges(g)
+1
+```
+"""
+function add_directed_edge!(mg::MixedGraph, source::Int64, target::Int64)
+  g = directed_component(mg)
+  _has_node(g, source) && _has_node(g, target) || return false
+  old_nedges = n_edges(g)
+  Polymake._add_edge(pm_object(g), source-1, target-1)
+  return n_edges(g) == old_nedges + 1
```

Comment:
```suggestion
  return add_edge!(directed_component(mg), source, target)
```
Wouldn't work this as well? If yes, IMO this is preferrable, as it reduces a lot of code duplication between the (Un)directed and the mixed cases.

Similar also for the other add and rem functions you introduce here